### PR TITLE
[Typing][B-21] Add type annotations for `python/paddle/distribution/multinomial.py`

### PR DESCRIPTION
**Pull request by enkilee**
### PR Category

User Experience

### PR Types

Improvements

### Description

- https://github.com/PaddlePaddle/Paddle/issues/65008
B21

### CODE REVIEW

File: python/paddle/distribution/multinomial.py [hunk 1]
```diff
@@ -66,7 +71,7 @@ class Multinomial(distribution.Distribution):
               [3., 3., 4.]]])
     """
 
-    def __init__(self, total_count, probs):
+    def __init__(self, total_count: int, probs: Tensor) -> None:
```
Comment (megemini) [inline python/paddle/distribution/multinomial.py:76]: 类属性 `total_count probs`

File: python/paddle/distribution/multinomial.py [hunk 2]
```diff
@@ -142,7 +147,7 @@ def log_prob(self, value):
             + (value * logits).sum(-1)
         )
 
-    def sample(self, shape=()):
+    def sample(self, shape: tuple[int]) -> Tensor:
```
Comment (megemini) [inline python/paddle/distribution/multinomial.py]:
```suggestion
    def sample(self, shape: Iterable[int]) -> Tensor:
```

用 `Iterable`  吧，下面代码用的这个做判断 ～

**Review by SigureMo — DISMISSED** [review]
LGTMeow

File: python/paddle/distribution/multinomial.py [hunk 3]
```diff
@@ -142,7 +149,7 @@ def log_prob(self, value):
             + (value * logits).sum(-1)
         )
 
-    def sample(self, shape=()):
+    def sample(self, shape: Iterable[int]) -> Tensor:
```
Comment (SigureMo) [inline python/paddle/distribution/multinomial.py]:
```suggestion
    def sample(self, shape: Iterable[int] = ()) -> Tensor:
```

**Review by SigureMo — APPROVED** [review]

**Pull request merged**